src/test/java/org/openhab/tools/analysis/checkstyle/test/PackageExportsNameCheckTest.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle.test;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.*;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openhab.tools.analysis.checkstyle.PackageExportsNameCheck;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheckTest;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Test for {@link PackageExportsNameCheck }
+ *
+ * @author Аleksandar Кovachev - Initial contribution
+ * @author Petar Valchev - Added a test for non existent packages
+ *
+ */
+public class PackageExportsNameCheckTest extends AbstractStaticCheckTest {
+    private static final String CORRECT_NAMING_OF_NOT_EXPORTED_PACKAGES_MESSAGE = "The package %s"
+            + " should be marked as \"internal\" if it will not be exported.";
+    private static final String NON_EXISTENT_PACKAGE_MESSAGE = "The package %s, that is exported from the MANIFEST.MF"
+            + " does not exist.";
+
+    private static final String MANIFEST_REALTIVE_PATH = META_INF_DIRECTORY_NAME + File.separator + MANIFEST_FILE_NAME;
+    private static final String TEST_DORECTORY = "packageExportsNameCheckTest";
+
+    private static DefaultConfiguration configuration;
+
+    @BeforeClass
+    public static void createConfiguration() {
+        configuration = createCheckConfig(PackageExportsNameCheck.class);
+        configuration.addAttribute("sourceDirectories", "src/main/java");
+        configuration.addAttribute("excludedPackages", ".*.internal.*");
+        configuration.addAttribute("excludedPackages", ".*.test.*");
+    }
+
+    @Test
+    public void testNotExportedPackages() throws Exception {
+        String manifestFilePath = getManifestPath("not_exported_packages");
+
+        int lineNumber = 0;
+        String[] expectedMessages = generateExpectedMessages(lineNumber,
+                String.format(CORRECT_NAMING_OF_NOT_EXPORTED_PACKAGES_MESSAGE, "org.openhab.not.exported.package"));
+
+        verify(configuration, manifestFilePath, expectedMessages);
+    }
+
+    @Test
+    public void testValidExports() throws Exception {
+        String manifestFilePath = getManifestPath("valid_exports");
+
+        String[] expectedMessages = CommonUtils.EMPTY_STRING_ARRAY;
+
+        verify(configuration, manifestFilePath, expectedMessages);

```

Comment:
See remark about this pattern in: https://github.com/openhab/static-code-analysis/pull/113/files#r133943368